File: projectEuler/6/6.py
#Find the difference between the sum of the squares of the first one hundred natural numbers and the square of the sum.

# The sum of the squares of the first n natural numbers is given by the formula n(n+1)(2n+1)/6
def sumOfSquares(n):
    return n*(n+1)*(2*n+1)/6
# The square of the sum of the first n natural numbers is given by the formula (n(n+1)/2)^2
def squareOfSum(n):
    return (n*(n+1)/2)**2
# The difference between the sum of the squares and the square of the sum is given by the difference of the two formulas
def difference(n):
    return squareOfSum(n) - sumOfSquares(n)
# Test the function with n = 100
print(difference(100))

File: projectEuler/6/6.cpp
#include <iostream>
using namespace std;

// Function to calculate sum of squares of first n natural numbers
long long sumOfSquares(int n) {
    return n * (n + 1) * (2 * n + 1) / 6;
}

// Function to calculate square of sum of first n natural numbers
long long squareOfSum(int n) {
    long long sum = n * (n + 1) / 2;
    return sum * sum;
}

// Function to find the difference between square of sum and sum of squares
long long difference(int n) {
    return squareOfSum(n) - sumOfSquares(n);
}

int main() {
    cout << difference(100) << endl;
    return 0;
}

/*
#include <iostream>
using namespace std;

// Function to find the difference between square of sum and sum of squares
long long difference(int n) {
    long long sum = n * (n + 1) / 2;
    long long sumSq = n * (n + 1) * (2 * n + 1) / 6;
    return sum * sum - sumSq;
}

int main() {
    cout << difference(100) << endl;
    return 0;
}
*/